packages/commonwealth/client/scripts/App.tsx

```diff
@@ -18,6 +21,12 @@ import { Mava } from './views/components/Mava';
 
 OpenFeature.setProvider(openFeatureProvider);
 
+// need to instantiate it early because the farcaster sdk has an async constructor which will cause a race condition
+// if instantiated right before the login is called;
+export const defaultMagic = new Magic(process.env.MAGIC_PUBLISHABLE_KEY!, {
```

Comment:
This might not be the best place to put it, but it needs to be instantiated when the app starts. Any better suggestions?

Reply:
not sure but does this cause circular dependency issues? or maybe we could add to `packages/commonwealth/client/scripts/controllers/app/login.ts` (that needs it - also login.ts is imported in many init flow conditions)?

Reply:
Yeah I moved it to login.ts. And yes this is the cause of the test failure issue. Not exactly a circular dependency but more of this Magic constructor calls the DOM, but our tests don't have the DOM so it throws an error. 

But basically this is needed because their constructor needs to complete an async request before we can call login. So basically the earlier it is instantiated the better, but still no guarantee that login won't fail when it is called.